ext/pdo_firebird/pdo_firebird_utils.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+  +----------------------------------------------------------------------+
+  | Copyright (c) The PHP Group                                          |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | https://www.php.net/license/3_01.txt                                 |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author: Simonov Denis <sim-mail@list.ru>                             |
+  +----------------------------------------------------------------------+
+*/
+
+#include "pdo_firebird_utils.h"
+#include <firebird/Interface.h>
```

Comment:
@sim1984 @cmb69 It seems that `firebird/Interface.h` isn't included in firebird 2.5: is it correct? If so, what about adding a note to the `UPGRADING` file, declaring the new firebird supported versions (from 3.0 to 5? Is it correct?)

Reply:
Uh, yes, this needs to be mentioned in these docs files also. So, the 3.0 is the minimum required now?

And then also version needs to be checked in config.m4.

Reply:
Oh right, I completely forgot about that. However, see https://github.com/php/php-src/pull/15230#issuecomment-2284635976. The question is: is fbclient 2.5 still relevant (i.e. used by any distros we support)?

Reply:
> is fbclient 2.5 still relevant (i.e. used by any distros we support)?

I don't think that on Alpine Linux there's a ready to use firebird client (at least, I couldn't find it).
So, the only solution is to manually compile it: when doing so, users can choose the firebird version they are going to install.

Reply:
Minimum requirements: fbclient version 3.0 or higher. Firebird server itself can be version 2.5, usually if it is on another machine.

Reply:
See PR #15498.